test/integration/serverless/log_normalize.py

```diff
@@ -0,0 +1,148 @@
+            #  /An error occurred while attempting to execute your code.: LambdaException/ or \
+            #  /terminate called after throwing an instance of 'std::logic_error'/ or \
+            #  /basic_string::_M_construct null not valid/" |
+    ]
+
+def normalize_traces(stage):
+    return [
+            require(r'BEGINTRACE.*ENDTRACE'),
+            exclude(r'BEGINTRACE'),
+            exclude(r'ENDTRACE'),
+            replace(r'(ts":)[0-9]{10}', r'\1XXX'),
+            replace(r'((startTime|endTime|traceID|trace_id|span_id|parent_id|start|system.pid)":)[0-9]+', r'\1null'),
+            replace(r'((tracer_version|language_version)":)["a-zA-Z0-9~\-\.\_]+', r'\1null'),
+            replace(r'(duration":)[0-9]+', r'\1null'),
+            replace(r'((datadog_lambda|dd_trace)":")[0-9]+\.[0-9]+\.[0-9]+', r'\1X.X.X'),
+            replace(r'(,"request_id":")[a-zA-Z0-9\-,]+"', r'\1null"'),
+            replace(r'(,"runtime-id":")[a-zA-Z0-9\-,]+"', r'\1null"'),
+            replace(r'(,"system.pid":")[a-zA-Z0-9\-,]+"', r'\1null"'),
+            replace(r'("_dd.no_p_sr":)[0-9\.]+', r'\1null'),
+            replace(r'("architecture":)"(x86_64|arm64)"', r'\1"XXX"'),
+            replace(r'("process_id":)[0-9]+', r'\1null'),
+            replace(stage, 'XXXXXX'),
+            exclude(r'[ ]$'),
+    ]
+
+#####################
+# BEGIN NORMALIZERS #
+#####################
+
+def replace(pattern, repl):
+    comp = re.compile(pattern, flags=re.DOTALL)
+    def _replace(log):
+        return comp.sub(repl, log)
+    return _replace
+
+def exclude(pattern):
+    comp = re.compile(pattern, flags=re.DOTALL)
+    return replace(pattern, '')
+
+def require(pattern):
+    comp = re.compile(pattern, flags=re.DOTALL)
+    def _require(log):
+        match = comp.search(log)
+        if not match:
+            return ''
+        return match.group(0)
+    return _require
+
+def sort_by(key):
+    def _sort(log):
+        log_json = json.loads(log, strict=False)
+        log_sorted = sorted(log_json, key=key)
+        return json.dumps(log_sorted)
+    return _sort
+
+###################
+# END NORMALIZERS #
+###################
+
+def normalize(log, typ, stage):
+    for normalizer in get_normalizers(typ, stage):
+        log = normalizer(log)
+    return format_json(log)
+
+def get_normalizers(typ, stage):
+    if typ == 'metrics':
+        return normalize_metrics(stage)
+    elif typ == 'logs':
+        return normalize_logs(stage)
+    elif typ == 'traces':
+        return normalize_traces(stage)
+    else:
+        raise ValueError(f'invalid type "{typ}"')
+
+def format_json(log):
+    if not log:
+        log = '{"error":"normalization returned empty payload"}'
+    try:
+        return json.dumps(json.loads(log, strict=False), indent=2)
+    except Exception as e:
+        print(json.dumps({
+            "error": "normalization raised exception: [{e.__class__.__name__}] {e}",
+            "normalized-logs": log,
+        }, indent=2))
```

Comment:
By returning a helpful error, it will be easier for us to debug problems with this code in the future.